python/cuml/neighbors/kneighbors_classifier.pyx

```diff
@@ -301,15 +294,15 @@ class KNeighborsClassifier(NearestNeighbors):
         cdef uintptr_t y_ptr
         for out_col in range(out_cols):
             col = self.y[:, out_col] if out_cols > 1 else self.y
-            classes = rmm.to_device(zeros((n_rows,
-                                           len(np.unique(np.asarray(col)))),
-                                          dtype=np.float32,
-                                          order="C"))
+            classes = CumlArray.zeros((n_rows,
+                                       len(cp.unique(cp.asarray(col)))),
```

Comment:
If this is to be done on device, we will need a `with_cupy_rmm` annotation on the `predict_proba` method.